domain/ide/workspace/folder_test.go

```diff
@@ -165,6 +167,57 @@ func TestProcessResults_whenSamePathsAndDuplicateIssues_DeDuplicates(t *testing.
 	assert.Len(t, f.documentDiagnosticCache.Get("path1"), 3)
 }
 
+func TestProcessResults_whenFilteringSeverity_ProcessesOnlyFilteredIssues(t *testing.T) {
+	testutil.UnitTest(t)
+
+	t.Run("Set severity settings", func(t *testing.T) {
+		config.SetCurrentConfig(config.New())
+		config.CurrentConfig().SetSeverityFilter(lsp.SeverityFilter{
+			Critical: true,
+			High:     false,
+			Medium:   true,
+			Low:      false,
+		})
+	})
+
+	f := NewFolder("dummy", "dummy", snyk.NewTestScanner(), hover.NewFakeHoverService())
+
+	f.processResults([]snyk.Issue{
+		{ID: "id1", AffectedFilePath: "path1", Severity: snyk.Critical},
+		{ID: "id2", AffectedFilePath: "path1", Severity: snyk.High},
+		{ID: "id3", AffectedFilePath: "path1", Severity: snyk.Medium},
+		{ID: "id4", AffectedFilePath: "path1", Severity: snyk.Low},
+		{ID: "id5", AffectedFilePath: "path1", Severity: snyk.Critical},
+	})
+
+	mtx := &sync.Mutex{}
+	var diagnostics []lsp.Diagnostic
+
+	defer notification.DisposeListener()
+	notification.CreateListener(func(event interface{}) {
```

Comment:
why not use the jsonRPCRecorder?

Reply:
I'm not familiar enough with it tbh and just followed the example set in the test below this one :)